### PR TITLE
Update to use nrf-pac instead of nrf-rs pacs.

### PR DESCRIPTION
**Pull request by Dirbaio**
Update to work with https://github.com/embassy-rs/embassy/pull/3498

- Code that doesn't do register access is changed to stable alternatives: `NVIC` from `cortex-m`, the `Interrupts` enum from `embassy-nrf`.
- Code that does raw register access is now done through `embassy_nrf::pac`. I think this is the better option even if it requires the `unstable-pac` feature, because it makes it easier to keep up to date with the PAC version. NVMC and timer regs are already pretty clean so it's unlikely we're going to get breakages there, while depending on the PAC directly means it'll need bumping the version here every time embassy-nrf bumps it.

Also as I suggested in https://github.com/embassy-rs/embassy/pull/3453 I've changed this to not require singletons for things that aren't in the HAL. (`nrf-pac` doesn't even have singletons now). If the HAL adds drivers for these in the future, we can change nrf-sdc to take the HAL singletons for them.

### CODE REVIEW

**Comment by Dirbaio** [comment]
btw could you do "repo settings -> actions -> general -> Require approval for first-time contributors who are new to GitHub"? this way new contributors don't have to wait for workflow approval.

**Comment by alexmoon** [comment]
This looks good to me. Since we can no longer rely on the singletons, can we document the peripherals that are off-limits to applications when using the MultiProtocolServiceLayer or SoftdeviceController?

**Comment by Dirbaio** [comment]
Done!

I've also added CCM to the list of forbidden peripherals, which was missing.

**Review by alexmoon — APPROVED** [review]

**Pull request merged**
